src/main/java/core/basesyntax/ArrayList.java

```
@@ -1,48 +1,127 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+
+    private int size = 0;
+    private int capacity = 10;
+    private Object[] array = new Object[capacity];
+
     @Override
     public void add(T value) {
-
+        try {
+            Objects.checkIndex(size, capacity);
+            array[size] = value;
+            size++;
+        } catch (IndexOutOfBoundsException e) {
+            grow();
+            add(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        try {
+            rangeCheck(index);
+            if (size == capacity) {
+                grow();
+            }
```

Error: The `rangeCheck(index)` method allows adding at any index up to `capacity`, but you should only allow adding at indices from 0 to `size` (inclusive). Inserting at an index greater than `size` should throw an exception, as per standard `ArrayList` behavior.